### PR TITLE
KAFKA-15823: disconnect from controller on AuthenticationException

### PR DESCRIPTION
**Pull request by gaurav-narula**
This PR changes the handling of authenticationException on a request from the node to the controller.

We disconnect controller connection and invalidate the cache so that the next run of the thread will establish a connection with the (potentially) updated controller.

### CODE REVIEW

**Review by OmniaGM — APPROVED** [review]
LGTM! I noticed the build seems to fail for an unrelated test. And the pipeline with `JDK 8 and Scala 2.12` crashed because of one of the stream test gradle tasks so maybe someone needs to kick this test again.

**Review by soarez — APPROVED** [review]
This change seems straightforward.

I think this not only solves the issue described in the JIRA, but also the case when a custom authenticaton store – either in either side, requestor or handler – is changed dynamically, causing failed authentication to be successful upon a retry.

**Comment by gaurav-narula** [comment]
CC: @hachikuji since you've looked into the response handler recently. Can you please review this PR?

**Comment by soarez** [comment]
This still needs a review from a committer. 

@mimaison @mumrah @rondagostino @showuon

File: core/src/main/scala/kafka/server/NodeToControllerChannelManager.scala
Comment (showuon) [inline core/src/main/scala/kafka/server/NodeToControllerChannelManager.scala:350]: Should we log: 
`Disconnecting the connection to the stale controller ${activeControllerAddress().map(_.idString).getOrElse("null")}`
like what we did for `NOT_CONTROLLER` error?

**Review by showuon — COMMENTED** [review]
LGTM! Just one minor comment. Thanks.

File: core/src/main/scala/kafka/server/NodeToControllerChannelManager.scala
Comment (gaurav-narula) [reply core/src/main/scala/kafka/server/NodeToControllerChannelManager.scala:350]: Thanks for the review! Addressed in [c405f73](https://github.com/apache/kafka/pull/14760/commits/c405f7387bcfc94845689d46588c9fc75cbb8596)

**Review by showuon — APPROVED** [review]
LGTM!

**Comment by showuon** [comment]
Failed tests are unrelated.

**Pull request merged**
